### PR TITLE
add focus change listener once

### PR DESCRIPTION
**Pull request by jrieken**
This PR is the minimal change I could come with to tackle #62970. It makes sure that only one focus change listener is added per (active) notification toast. That prevent the exponential listener growth/leak we have observed

### CODE REVIEW

**Review by joaomoreno — APPROVED** [review]
Great minimal fix. It's hard to follow what this class is doing (it's not really `purgeNotification` but rather `eventuallyMaybePurgeNotification`, etc), but the fix is solid, given the repro steps we've got.

**Comment by jrieken** [comment]
Thanks @joaomoreno
